bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -185,13 +187,28 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, cluster *clusterv1
 
 	log.Info("Generating userdata")
 
-	// generate userdata
-	userDataScript, err := userdata.NewNode(&userdata.NodeInput{
+	nodeInput := &userdata.NodeInput{
 		// AWSManagedControlPlane webhooks default and validate EKSClusterName
-		ClusterName: controlPlane.Spec.EKSClusterName,
-
+		ClusterName:      controlPlane.Spec.EKSClusterName,
 		KubeletExtraArgs: config.Spec.KubeletExtraArgs,
-	})
+		ContainerRuntime: config.Spec.ContainerRuntime,
+		DNSClusterIP:     config.Spec.DNSClusterIP,
+		DockerConfigJSON: config.Spec.DockerConfigJSON,
+		APIRetryAttempts: config.Spec.APIRetryAttempts,
+		UseMaxPods:       config.Spec.UseMaxPods,
+	}
+	if config.Spec.PauseContainer != nil {
+		nodeInput.PauseContainerAccount = &config.Spec.PauseContainer.AccountNumber
+		nodeInput.PauseContainerVersion = &config.Spec.PauseContainer.Version
+	}
+	// TODO(richardcase): uncomment when we support ipv6 / dual stack
+	/*if config.Spec.ServiceIPV6Cidr != nil && *config.Spec.ServiceIPV6Cidr != "" {
+		nodeInput.ServiceIPV6Cidr = config.Spec.ServiceIPV6Cidr
+		nodeInput.IPFamily = pointer.String("ipv6")
```

Comment:
For future implementation: 
If setting `IPFamily`  without setting `ServiceIPV6Cidr` possible during EKS bootstrap, then I think having `IPFamily` as a separate field is needed.

Reply:
It is possible to set `IPFamily` without setting `ServiceIPV6Cidr` but i think we should hide this detail based on whether the user has supplied ipv4 or 6